subsys/net/lib/sntp/sntp_simple.c

```diff
@@ -65,3 +68,22 @@ int sntp_simple(const char *server, uint32_t timeout, struct sntp_time *time)
 
 	return res;
 }
+
+#if defined(CONFIG_POSIX_CLOCK)
+int sntp_set_system_time(const char *server, uint32_t timeout)
+{
+	struct sntp_time ts;
+	struct timespec tspec;
+	int res = sntp_simple(server, timeout, &ts);
+
+	if (res < 0) {
+		return res;
+	}
+
+	tspec.tv_sec = ts.seconds;
+	tspec.tv_nsec = ((uint64_t)ts.fraction * (1000 * 1000 * 1000)) >> 32;
```

Comment:
I wonder if this is correct, but I see that you just copied it.
I think this does some some assumptions on the architecture.

Reply:
Why would it. These are two defined data types, the one of the Posix system, the other the response from a SNTP query. I guess this has nothing to do with the architecture, but rather a conversion between these two data types.

Reply:
No, this is correct, silly me